Robust.Client/UserInterface/CustomControls/DebugConsole.cs

```diff
@@ -204,6 +212,83 @@ private void CommandBarOnOnKeyBindDown(GUIBoundKeyEventArgs args)
                 Output.ScrollToBottom();
                 args.Handle();
             }
+            else if (args.Function == EngineKeyFunctions.GuiTabNavigateNext)
+            {
+                NextCommand();
+                args.Handle();
+                return;
+            }
+            else if(args.Function == EngineKeyFunctions.GuiTabNavigatePrev)
+            {
+                PrevCommand();
+                args.Handle();
+                return;
+            }
+        }
+
+        private void SetInput(string cmd)
+        {
+            CommandBar.Text = cmd;
+            CommandBar.CursorPosition = cmd.Length;
+        }
+        
+        private void FindCommands()
+        {
+            searchResults.Clear();
+            searchIndex = 0;
+            commandChanged = false;
+            foreach (var cmd in Commands)
+            {
+                if (cmd.Key.StartsWith(CommandBar.Text))
+                {
+                    searchResults.Add(cmd.Key);
+                }
+            }
+        }
+
+        private void NextCommand()
+        {
+            if (!commandChanged)
+            {
+                if (searchResults.Count == 0)
+                    return;
+
+                searchIndex = ++searchIndex % searchResults.Count;
+                SetInput(searchResults[searchIndex]);
+                return;
+            }
+
+            FindCommands();
+            if (searchResults.Count == 0)
+                return;
+
+            SetInput(searchResults[0]);
+        }
+
+        private void PrevCommand()
+        {
+            if (!commandChanged)
+            {
+                if (searchResults.Count == 0)
+                    return;
+
+                searchIndex--;
+                if (searchIndex < 0)
+                    searchIndex += searchResults.Count;
+                SetInput(searchResults[searchIndex]);
+                return;
+            }
+
+            FindCommands();
+            if (searchResults.Count == 0)
+                return;
+
+            SetInput(searchResults[searchResults.Count - 1]);
```

Comment:
You can use C# 8 `Index`es.

```suggestion
            SetInput(searchResults[^1]);
```